src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -378,24 +375,45 @@ export class StudyViewPageStore
     @observable userGroupColors: { [groupId: string]: string } = {};
 
     @action
-    updateNAValue = (uniqueKey: string): void => {
-        let newValue = this.isShowNAChecked(uniqueKey);
-        let newFilter = _.clone(this._clinicalDataBinFilterSet.get(uniqueKey))!;
+    updateNAValue = (
+        uniqueKey: string,
+        dataType: ChartMetaDataTypeEnum
+    ): void => {
+        let newValue = this.isShowNAChecked(uniqueKey, dataType);
+        const filterSet =
+            dataType === ChartMetaDataTypeEnum.GENE_SPECIFIC
```

Comment:
for some reason i hate that we have to do this check and then repeat it in some many places.  like, it feels to that this kind of function shouldn't need knowledge of the type.  it should instead reside in the chart obj itself?